File: WoofAndroid/domain/usecase/src/main/kotlin/com/perrystreet/woof/usecase/dogs/GetDogProfileUseCase.kt
package com.perrystreet.woof.usecase.dogs

import com.perrystreet.woof.models.dog.Dog
import com.perrystreet.woof.models.dog.DogProfile
import com.perrystreet.woof.repositories.dogs.DogsRepository
import io.reactivex.rxjava3.core.Single
import org.koin.core.annotation.Factory

@Factory
class GetDogProfileUseCase(
    private val dogsRepository: DogsRepository,
) {
    operator fun invoke(dog: Dog): Single<DogProfile> = dogsRepository.getDogProfile(dog)
}

File: WoofIOS/SwiftPackages/Domain/UseCase/Sources/UseCase/Dogs/GetDogProfileUseCase.swift
import Combine
import DI
import Models
import Repositories

@Factory
public final class GetDogProfileUseCase {
    private let dogsRepository: DogsRepository

    public func callAsFunction(dog: Dog) -> AnyPublisher<DogProfile, DogsError> {
        dogsRepository.getDogProfile(dog: dog)
    }
}
